lib/Target/RISCV/MCTargetDesc/RISCVMCTargetDesc.cpp

```diff
@@ -59,6 +59,32 @@ static MCInstPrinter *createRISCVMCInstPrinter(const Triple &T,
   return new RISCVInstPrinter(MAI, MII, MRI);
 }
 
+/// Select the RISCV CPU for the given triple and cpu name.
+StringRef RISCV_MC::selectRISCVCPU(const Triple &TT, StringRef CPU) {
```

Comment:
The way this currently works, we'd need to define a different CPU for every RISC-V variant. It would be better to instead modify the featurestring based on parsing the arch.

Reply:
Yes, I think you're right. So I add parseRISCVTripple to get feature string instead of using CPU.